X265 multilib

jstebbins opened the pull request.

sr55 committed:
Refactoring the Preset service to store the Built-In Presets and User Presets in the new version of the JSON format. This allows HandBrakeCLI to use the presets stored by the GUI.

sr55 committed:
Fixing the storage of user presets. Need to confirm PresetContainer format.

jstebbins committed:
x265: use multi-lib API

This does not yet support other bit depths.  But it's the first step to
doing so.

jstebbins committed:
x265: use multi-lib API

This does not yet support other bit depths.  But it's the first step to
doing so.

jstebbins committed:
x265: add support for 10, 12, and 16 bit x265 libs

automatically pull in shared versions of these libs or allow statically
building against any one of them.

jstebbins committed:
x265: add support for 10, 12, and 16 bit x265 libs

automatically pull in shared versions of these libs or allow statically
building against any one of them.

bradleysepos commented on libhb/common.c:
```diff
@@ -218,19 +221,22 @@ hb_encoder_t *hb_video_encoders_last_item  = NULL;
+    { { "H.265 8 bit (x265)",  "x265",       "H.265 8 bit (libx265)",   HB_VCODEC_X265_8BIT,      HB_MUX_AV_MP4|HB_MUX_AV_MKV,   }, NULL, 1, HB_GID_VCODEC_H265_8BIT,   },
+    { { "H.265 10 bit (x265)", "x265_10bit", "H.265 10 bit (libx265)",  HB_VCODEC_X265_10BIT,     HB_MUX_AV_MP4|HB_MUX_AV_MKV,   }, NULL, 1, HB_GID_VCODEC_H265_10BIT,   },
+    { { "H.265 12 bit (x265)", "x265_12bit", "H.265 12 bit (libx265)",  HB_VCODEC_X265_12BIT,     HB_MUX_AV_MP4|HB_MUX_AV_MKV,   }, NULL, 1, HB_GID_VCODEC_H265_12BIT,   },
+    { { "H.265 16 bit (x265)", "x265_16bit", "H.265 16 bit (libx265)",  HB_VCODEC_X265_16BIT,     HB_MUX_AV_MP4|HB_MUX_AV_MKV,   }, NULL, 1, HB_GID_VCODEC_H265_16BIT,   },
```
Hyphen all the things.

ghost commented:
I see no build changes, I gather this will only for with system libs for now, we don't include dynamic libraries in the package and help x265 find them yet?

Also generally, I think it's better to have the bit depth not be a specific encoder, but a drop-down on the video tab.

Then again we introduce more GUI complexity because to be correct, we'll need to update the profile list every time we change bit depth :(

jstebbins committed:
LinGui: Fix "More settings" box dependencies

jstebbins committed:
LinGui: Fix "More settings" box dependencies

jstebbins committed:
x265: encoder name cosmetics "X bit" -> "X-bit"

jstebbins committed:
x265: encoder name cosmetics "X bit" -> "X-bit"

jstebbins committed:
x265: cleaner line breaking cosmetics

jstebbins committed:
x265: cleaner line breaking cosmetics

jstebbins committed:
x265: fix high bit depth profiles

jstebbins committed:
x265: fix high bit depth profiles

jstebbins commented:
Correct, there are no build changes.  This only works if the user manually installs their own shared libraries following the instructions for multilib support https://x265.readthedocs.org/en/1.7/api.html#multi-library-interface

Adding a bit-depth field would require modifying 3 guis and adding a new preset key.  We can certainly do that, but it would require much more coordination and time.  For a feature that is only available to a few users that know what they are doing to install the multi-libs, it seems like too much work.

ghost commented:
What if we figure out how to make (+10-bit work for x264 and/or figure out how to ship our own libs with HB? Might be less of a niche feature then :)

jstebbins committed:
x265: fix compile error

missed a use of hb_x265_profile_name

jstebbins committed:
x265: fix compile error

missed a use of hb_x265_profile_name

jstebbins commented:
Figuring out distribution of shared libs and build changes to build multiple versions of the libraries is a whole other patch.  Your just making this less likely to happen by piling on.

ghost commented:
Oh, I missed the new patches, BTW. Well, if we're going to go with one encoder per bit depth, patch(set) LGTM at this point :)

bradleysepos commented:
Nit: H.265 8-bit (x265) can be presented to the user as H.265 (x265). 8-bit is assumed across all codecs at this point.

jstebbins committed:
LinGui: Put filters in their own pane

And rearage their order to follow the order they are applied in during
encoding.

galad87 committed:
MacGui: added accessibility text to the preferences popup buttons.

jstebbins committed:
LinGui: Fix filter tab on gtk 3.10

Gtk versions prior to 3.12 do not support GtkFlowBox

galad87 committed:
MacGui: align the presets view to the top

galad87 committed:
MacGui: add a test for custom anamorphic

galad87 committed:
MacGui: cleanup the csv import/export code and the chapters view controller.

galad87 committed:
MacGui: do not try to import a preset even if the cancel button was clicked.

jstebbins committed:
LinGui: fix "margin-start" run-time warning on gtk 3.10

sr55 committed:
Completely re-factored the preset file storage to use the same format as the Mac and Linux GUI's. You can now copy the file between platforms and it should load correctly.

WARNING: Any current presets will be lost when you install this or any later build as the format is not backwards compatible and there is no upgrade path.

jstebbins committed:
Merge pull request #20 from jstebbins/mux-merge

mux: flush cached buffers when closing

galad87 committed:
MacGui: renamed the built-in preset menu item to 'Reset Built-in Presets'.

galad87 committed:
MacGui: remove the Base.lproj folder because base localizations require 10.8, and we still support 10.7.

galad87 committed:
Remove obsolete InstantHandBrake project.

jstebbins committed:
x265: multilib fix CLI x265 encoding

... and other clean-ups

jstebbins committed:
x265: macui multilib support

jstebbins committed:
Merge remote-tracking branch 'origin/x265-multilib' into x265-multilib
